servers/rendering/renderer_canvas_cull.cpp

```diff
@@ -232,14 +232,22 @@ void RendererCanvasCull::_attach_canvas_item_for_draw(RendererCanvasCull::Item *
 	}
 }
 
-void RendererCanvasCull::_cull_canvas_item(Item *p_canvas_item, const Transform2D &p_parent_xform, const Rect2 &p_clip_rect, const Color &p_modulate, int p_z, RendererCanvasRender::Item **r_z_list, RendererCanvasRender::Item **r_z_last_list, Item *p_canvas_clip, Item *p_material_owner, bool p_allow_y_sort, uint32_t p_canvas_cull_mask, const Point2 &p_repeat_size, int p_repeat_times) {
+void RendererCanvasCull::_cull_canvas_item(Item *p_canvas_item, const Transform2D &p_parent_xform, const Rect2 &p_clip_rect, const Color &p_modulate, int p_z, RendererCanvasRender::Item **r_z_list, RendererCanvasRender::Item **r_z_last_list, Item *p_canvas_clip, Item *p_material_owner, bool p_allow_y_sort, uint32_t p_canvas_cull_mask, const Point2 &p_repeat_size, int p_repeat_times, uint32_t p_parent_visibility_layer) {
 	Item *ci = p_canvas_item;
 
 	if (!ci->visible) {
 		return;
 	}
 
-	if (!(ci->visibility_layer & p_canvas_cull_mask)) {
+	uint32_t visibility_layer = ci->visibility_layer;
+
+	if (ci->inherit_visibility_layer) {
+		if (!(p_parent_visibility_layer & p_canvas_cull_mask)) {
+			return;
+		}
+
+		visibility_layer = p_parent_visibility_layer;
+	} else if (!(ci->visibility_layer & p_canvas_cull_mask)) {
 		return;
 	}
```

Comment:
Seems simpler to me:
```suggestion
	uint32_t visibility_layer = ci->inherit_visibility_layer ? p_parent_visibility_layer : ci->visibility_layer;
	if (!(visibility_layer & p_canvas_cull_mask)) {
		return;
	}
```